lib/asciidoctor-epub3/packager.rb

```diff
@@ -446,20 +447,20 @@ def package options = {}
     end
 
     if fmt == :kf8
-      distill_epub_to_mobi epub_file
+      distill_epub_to_mobi epub_file, target
     elsif options[:validate]
       validate_epub epub_file
     end
   end
 
   # QUESTION how to enable the -c2 flag? (enables ~3-5% compression)
-  def distill_epub_to_mobi epub_file
+  def distill_epub_to_mobi epub_file, target
     kindlegen_cmd = KINDLEGEN
     unless ::File.executable? kindlegen_cmd
       require 'kindlegen' unless defined? ::Kindlegen
       kindlegen_cmd = ::Kindlegen.command
     end
-    mobi_file = ::File.basename(epub_file).sub Kf8ExtensionRx, '.mobi'
+    mobi_file = ::File.basename(target)
```

Comment:
When I set `-o myname.mobi`, it doesn't need to change filename here. For example:

`-o myname.mobi`, epub_file is `/path/to/myname-kf8.epub`, target is `/path/to/myname.mobi`, mobi_file is `myname.mobi`, it's the same whether `.sub Kf8ExtensionRx, '.mobi'` here.

And if target is not ending with `.mobi`, I think it should not change it too, just let user control it.

`Kf8ExtensionRx` is not being used in other places, I think it should be delete.